### PR TITLE
Refactored DfsLogger constructors to static methods that now call a p…

### PR DESCRIPTION
**Pull request by ArbaazKhan1**
I made the dfslogger constructors private and now have them being called from static methods. To ensure that LogEntry can now be set as final in dfslogger, I removed the instantiation of it from the open() method and now have it initialized it in the newly created static method named fromCounters().

### CODE REVIEW

File: server/tserver/src/main/java/org/apache/accumulo/tserver/log/TabletServerLogger.java [hunk 1]
```diff
@@ -266,7 +266,8 @@ private synchronized void startLogMaker() {
         DfsLogger alog = null;
 
         try {
-          alog = new DfsLogger(tserver.getContext(), syncCounter, flushCounter);
+          alog = DfsLogger.fromCounters(tserver.getContext(), syncCounter, flushCounter,
+              tserver.getClientAddressString());
           alog.open(tserver.getClientAddressString());
```
Comment (DomGarguilo) [inline server/tserver/src/main/java/org/apache/accumulo/tserver/log/TabletServerLogger.java:271]: This is the spot referenced in the ticket and I think can be addressed by my other comments.

File: server/tserver/src/main/java/org/apache/accumulo/tserver/log/DfsLogger.java [hunk 2]
```diff
@@ -291,16 +291,36 @@ public int hashCode() {
...
+
+    return new DfsLogger(context, syncCounter, flushCounter, LogEntry.fromPath(logPath));
+  }
+
+  public static DfsLogger fromExistingLogEntry(ServerContext context, LogEntry logEntry) {
+    return new DfsLogger(context, logEntry);
+  }
+
+  private DfsLogger(ServerContext context, AtomicLong syncCounter, AtomicLong flushCounter,
+      LogEntry logEntry) {
```
Comment (DomGarguilo) [inline server/tserver/src/main/java/org/apache/accumulo/tserver/log/DfsLogger.java:322]: This is pretty similar to another approach that I thought of that makes it so we can call open within the `fromCounters` method.

What we might be able to do is make a single private constructor that sets all of the fields that both of these static methods use. The `fromExistingLogEntry` is more straight forward where we can just call the private constructor but the `fromCounters` we could do the following:

After creating the `String logPath`, create the `LogEntry` and use all of the variables to create a DfsLogger object to then call `open` on. Then return that DfsLogger. Like this:
```java
    String logPath = fs.choose(chooserEnv, context.getBaseUris()) + Path.SEPARATOR
        + Constants.WAL_DIR + Path.SEPARATOR + loggerAddress + Path.SEPARATOR + filename;

    LogEntry logEntry = LogEntry.fromPath(logPath);
    DfsLogger logger = new DfsLogger(context, logEntry, syncCounter, flushCounter);

    logger.open(fs, logPath, filename);
    return logger;
```

That way we can make open private too and LogEntry can be final.

File: server/tserver/src/main/java/org/apache/accumulo/tserver/log/DfsLogger.java [hunk 3]
```diff
@@ -291,16 +291,40 @@ public int hashCode() {
...
   private Thread syncThread;
 
   private AtomicLong syncCounter;
   private AtomicLong flushCounter;
   private final long slowFlushMillis;
   private long writes = 0;
 
-  public DfsLogger(ServerContext context, AtomicLong syncCounter, AtomicLong flushCounter) {
-    this(context, null);
+  public static DfsLogger fromCounters(ServerContext context, AtomicLong syncCounter,
```
Comment (DomGarguilo) [inline server/tserver/src/main/java/org/apache/accumulo/tserver/log/DfsLogger.java:302]: I'm not sure of the best name for this method but maybe something like `fromNewLogEntry` might be a good fit too. I am also fine with leaving it as is so unless someone else has an opinion I think its fine.

**Review by DomGarguilo — APPROVED** [review]

File: server/tserver/src/main/java/org/apache/accumulo/tserver/log/DfsLogger.java (hunk 3 above)
Comment (ctubbsii) [reply server/tserver/src/main/java/org/apache/accumulo/tserver/log/DfsLogger.java]: I pushed a different name in my code review commit, in collab with @ArbaazKhan1

**Comment by ctubbsii** [comment]
Thanks for the PR @ArbaazKhan1. I see you aren't listed on our "people" page, acknowledging contributors. If you wish to be added as a contributor to https://accumulo.apache.org/people/ , please open a pull request to add yourself at https://github.com/apache/accumulo-website/edit/main/pages/people.md and leave a reference to `apache/accumulo#4192` in your commit log.

If you intend to be a regular contributor to Accumulo projects, please consider subscribing to our developer mailing list (https://accumulo.apache.org/contact-us/) and introducing yourself. :smiley_cat:

**Pull request merged**